openapi_server/apis/reviews_api.py

```diff
@@ -42,7 +54,88 @@ async def add_review(
     ),
 ) -> Review:
     """Add a new review about a restaurant"""
-    ...
+    try:
+        user: DbUser = DbUser.get(create_review.username)
+    except DbUser.DoesNotExist:
+        raise HTTPException(status_code=404, detail="User not found")
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+    # check if the restaurant information is stored in the database; create the item if not
+    try:
+        restaurant: DbRestaurant = DbRestaurant.get(create_review.restaurant_id)
+    except DbRestaurant.DoesNotExist:
+        try:
+            async with AsyncClient(base_url="https://maps.googleapis.com") as ac:
+                response = await ac.get(
+                    f"/maps/api/place/details/json?place_id={create_review.restaurant_id}&key={os.environ.get('GOOGLE_API_KEY')}"
+                )
+                result: Dict = response.json()["result"]
+
+            restaurant = DbRestaurant(result["place_id"])
+            restaurant.update(
+                actions=[
+                    DbRestaurant.name.set(result["name"]),
+                    DbRestaurant.latitude.set(result["geometry"]["location"]["lat"]),
+                    DbRestaurant.longitude.set(result["geometry"]["location"]["lng"]),
+                    DbRestaurant.address.set(result["formatted_address"]),
+                ]
+            )
+            restaurant.save()
```

Comment:
Doesn't it perform two operations in DB? update and save? we could just assign the properties and save such as
`restaurant.name = result["name"]
...
restaurant.save()`

or might be possible to set the value at once 
`restaurant = DbRestaurant(id=result["place_id"], name=result["name], ...)`

Reply:
After some testing, it seems like you are correct. It is possible to set the values all at once upon item creation. Since item creation takes the second parameter as the range key and there is no RK currently set up for DbRestaurant, I had thought this method wouldn't work. I should have tried specifying the assignment operator even as the second parameter. The change will look something like `restaurant = DbRestaurant(result["place_id"], name=result["name"], ...)` at line 75 and just keeping line 84. `users_api.py` will need an optimization like this as well.